bundles/org.openhab.binding.publictransportswitzerland/src/main/java/org/openhab/binding/publictransportswitzerland/internal/stationboard/PublicTransportSwitzerlandStationboardHandler.java

```diff
@@ -71,8 +72,15 @@ public PublicTransportSwitzerlandStationboardHandler(Thing thing) {
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
         if (command instanceof RefreshType) {
-            stopDataUpdate();
-            startDataUpdate();
+            long currentTime = System.currentTimeMillis();
+
+            // Avoid hitting API limits
+            if (currentTime - lastRefreshTime > 30_000) {
+                lastRefreshTime = currentTime;
+
+                stopDataUpdate();
+                startDataUpdate();
+            }
```

Comment:
In fact, the good approach would be to use a cache expiry like in few other bindings. You will cache the result of your request. Like that, every REFRESH command will be handled without invoking each time the request.